mne/preprocessing/xdawn.py

```diff
@@ -261,26 +252,25 @@ def fit(self, epochs, y=None):
 
         # estimates evoked covariance
         self.evokeds_cov_ = dict()
-        if self.correct_overlap:
-            if epochs.baseline is not None:
-                raise ValueError('Baseline correction must be None if overlap '
-                                 'correction activated')
-            evokeds, toeplitz = least_square_evoked(epochs,
-                                                    return_toeplitz=True)
-        else:
-            evokeds = dict()
-            toeplitz = dict()
-            for eid in epochs.event_id:
-                evokeds[eid] = epochs[eid].average()
-                toeplitz[eid] = 1.0
+        evokeds = dict()
+        toeplitz = dict()
+        classes = np.unique(y)
+        shape = epochs_data.shape
+        for eid in classes:
+            mean_data = np.mean(epochs_data[eid].reshape(-1, shape[0],
+                                                            shape[1]),
+                                    axis=0)
+            picks = _pick_data_channels(self.info, exclue=[])
```

Comment:
This channel picking should probably be handled by EpochsVectorizer. @dengemann WDYT?